drivers/spi/spi_ll_stm32.c

```diff
@@ -338,20 +338,36 @@ static int spi_stm32_get_err(SPI_TypeDef *spi)
 	return 0;
 }
 
-/* Shift a SPI frame as master. */
-static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
+static void spi_stm32_shift_fifo(SPI_TypeDef *spi, struct spi_stm32_data *data)
 {
-	while (!ll_func_tx_is_not_full(spi)) {
-		/* NOP */
+	if (ll_func_rx_is_not_empty(spi)) {
+		spi_stm32_read_next_frame(spi, data);
 	}
 
-	spi_stm32_send_next_frame(spi, data);
-
-	while (!ll_func_rx_is_not_empty(spi)) {
-		/* NOP */
+	if (ll_func_tx_is_not_full(spi) && spi_stm32_transfer_ongoing(data)) {
```

Comment:
Is `spi_stm32_transfer_ongoing(data)` necessary here? As the check is done before in 
https://github.com/zephyrproject-rtos/zephyr/blob/1d61642388097d4afcd567c3043a5829e27a2f14/drivers/spi/spi_ll_stm32.c#L530